truenas_install/__main__.py

```diff
@@ -235,17 +246,68 @@ def configure_system_for_zectl(boot_pool):
         run_command(["zfs", "set", "org.zectl:bootloader=grub", root_ds])
 
 
+def read_license(root):
+    license_path = os.path.join(root, "data/license")
+    try:
+        with open(license_path) as f:
+            return License.load(f.read().strip('\n'))
+    except Exception:
+        return None
+
+
+def precheck(old_root):
+    if old_root is not None:
+        if licenseobj := read_license(old_root):
+            if ContractType(licenseobj.contract_type) in [ContractType.silver, ContractType.gold]:
+                if licenseobj.contract_end > datetime.utcnow().date():
+                    db_path = database_path(old_root)
+                    if os.path.exists(db_path):
+                        try:
+                            if query_row(
+                                "SELECT * FROM services_services WHERE srv_service = 's3' AND srv_enable = 1",
+                                db_path,
+                            ) is not None:
+                                return (
+                                    "You have built-in S3 service enabled in your configuration. This is not supported "
+                                    "anymore. Please, contact our support team."
+                                )
+                        except Exception:
+                            pass
+
+                    for p in psutil.process_iter():
+                        if p.name() == "minio":
+                            try:
+                                with open(f"/proc/{p.pid}/cgroup") as f:
+                                    cgroups = f.read()
+                            except FileNotFoundError:
+                                cgroups = ""
+
+                            if "kubepods.slice" not in cgroups:
```

Comment:
@sonicaj is this a good method to check whether the `minio` process is running as an app?

Reply:
Yes this should be a good way to identify that